addons/mail/models/mail_channel.py

```diff
@@ -92,6 +92,7 @@ def _generate_random_token(self):
                                       default=lambda self: self.env.ref('base.group_user'))
 
     _sql_constraints = [
+        ('channel_type_not_null', 'CHECK(channel_type IS NOT NULL)', 'The channel type cannot be empty'),
```

Comment:
Ideally we should ensure the value is actually in the selection range rather than just not empty, but as the possibilities change dynamically when other modules are installed that's a pain to handle. Out of scope of this PR but no idea why we have to make such constraints manually though, I wish the ORM just did it for us on required/selection fields.